app/src/main/java/com/github/polybooks/core/database/DatabaseAPI.kt

```diff
@@ -0,0 +1,176 @@
+package com.github.polybooks.core.database
+
+import com.github.polybooks.core.*
+import java.util.concurrent.CompletableFuture
+
+/**
+ * Provides the API for a database. Allows to retrieve information in the database
+ * about Books, Sales.
+ * */
+interface Database {
+
+    /**
+     * Creates a new query for Books. It originally matches all books.
+     * */
+    fun queryBooks() : BookQuery
+
+    /**
+     * Create a new query for Sales. It originally matches all sales.
+     * */
+    fun querySales() : SaleQuery
+
+    //TODO we could or could not include those functions (and more) in the interface. Should we?
+
+    fun listAllBooks() : CompletableFuture<List<Book>> = TODO("It can be implemented from the previous functions")
+
+    fun listAllSales() : CompletableFuture<List<Sale>> = TODO("It can be implemented from the previous functions")
+
+    /**
+     * Get data about a Book from the database given it's ISBN13
+     * */ //TODO define how to handle an invalid isbn13
```

Comment:
Isn't that part of the Futures features?

Reply:
Futures can be completed exceptionally, but the interface should define in more details what exceptions are "thrown" in which case.